src/auditor/soname_matching.jl

```diff
@@ -47,6 +47,19 @@ function ensure_soname(prefix::Prefix, path::AbstractString, platform::AbstractP
         return true
     end
 
+    # macOS uses install_name_tool (check first since Sys.isbsd is true for macOS too)
+    if Sys.isapple(platform)
+        return _ensure_soname_macho(prefix, path, platform; verbose, autofix, subdir)
+    end
+
+    # For Linux/FreeBSD, wrap the entire read-modify-verify cycle in the file lock
+    return with_patchelf_lock(path) do
+        _ensure_soname_elf(prefix, path, platform; verbose, autofix, subdir)
+    end
+end
+
+function _ensure_soname_macho(prefix::Prefix, path::AbstractString, platform::AbstractPlatform;
+                              verbose::Bool = false, autofix::Bool = false, subdir::AbstractString="")
```

Comment:
The fact that this function takes `autofix` as argument is code smell: we shouldn't call this function to start with if `autofix`is false.